src/ploomber_core/exceptions.py

```diff
@@ -75,6 +75,21 @@ def __init__(self, message):
         super().__init__(f"{message}. {COMMUNITY.strip()}")
 
 
+class ValidationError(BaseException):
+    """Raised when failed to validate input data
+    """
+    pass
+
+
+class MissingKeysValidationError(ValidationError):
+    """Raised when failed to validate input data because keys were missing
+    """
+
+    def __init__(self, message, missing_keys):
+        self.missing_keys = missing_keys
```

Comment:
what is the purpose of missing keys? is it used at all?

Reply:
I'm not sure, I took it from the old code in Ploomber and I'm not using the `MissingKeysValidationError`.
I went over the Ploomber codebase and it seems like it doesn't do anything except initialize `missing_keys` when raising the error.

```python
if missing:
    raise MissingKeysValidationError(
        f"Error validating {name}. Missing "
        f"keys: { pretty_print.iterable(missing)}",
        missing_keys=missing)
```